src/pilot.c

```diff
@@ -2903,6 +2909,14 @@ void pilot_destroy(Pilot* p)
 
    /* copy other pilots down */
    memmove(&pilot_stack[i], &pilot_stack[i+1], (pilot_nstack-i)*sizeof(Pilot*));
+
+   /* Clear docks for launched fighters if necessary */
+   for (i=0; i < pilot_nstack; i++) {
```

Comment:
I think this will work...

But it means every `pilot_destroy` call iterates over every pilot in the system, which might not be the most efficient solution.

If instead `dockpilot` is a pilot ID, and `dockslot` an index in the slots, I think that would be both more efficient and easier not to break (since there aren't pointers lying around to potentially become dangling).

Reply:
I think using a pilot ID and slot index should work nicely with a helper function like this (untested):

```c
PilotOutfitSlot* pilot_getDockSlot( Pilot* p ) {
   Pilot* dockpilot;
   if (p->dockpilot != 0) {
      docpilot = pilot_get(p->dockpilot);
      if (docpilot == NULL)
         p->docpilot = 0;
      else
         return docpilot->outfits[p->dockslot];
   }
   return NULL;
}
```

Reply:
That's real fair. It did occur to me that would be cleaner as an implementation, too.

Reply:
OK so I'm finding implementing it that way to be a bit cumbersome. Giving up on that for now, I don't have the energy for it.

(Also worth noting, that change isn't going to reduce looping at all, the looping is to find the ships that mark the pilot being destroyed as a dock and changing the way said pilot is marked won't prevent having to loop to do that. The only way to avoid that is if we use a list stored for each pilot that shows all the pilots that use them as a dock, which I guess the escort list *might* work for, but I haven't looked into that yet.)

Reply:
> that change isn't going to reduce looping at all

With the way I suggested, I *think* this loop could be removed. When the escort's code tries to call `pilot_getDockSlot()`, that will simply return `NULL` since the pilot that launched it wouldn't exist any more; without `docpilot`/`docslot` having to be updated already.

Reply:
Alright, as per usual after whining for a bit I did the conversion. It's using ints for both dockpilot and dockslot now.